addons/content-index/src/main/java/org/commonjava/indy/content/index/ISPKey2StringMapper.java

```diff
@@ -0,0 +1,88 @@
+/**
+ * Copyright (C) 2013 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.indy.content.index;
+
+import org.infinispan.commons.util.Base64;
+import org.infinispan.persistence.keymappers.TwoWayKey2StringMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+
+public class ISPKey2StringMapper
```

Comment:
Seems like this implementation will perform better than the StringBuilder-based one. Since you're base-64 encoding in both cases, it's not like we can do substring searches directly against the database in any case, so I'd say going with the fastest solution is likely best.